src/include/platform/DeviceInfoProvider.h

```diff
@@ -29,7 +29,7 @@
 namespace chip {
 namespace DeviceLayer {
 
-static constexpr size_t kMaxUserLabelListLength = 10;
+static constexpr size_t kMaxUserLabelListLength = 4;
```

Comment:
Why this change?  Spec says "at least 4"....

Reply:
Because right now everyone is forced to be able to store up to 10 and it seems like a leak in waiting.